src/setRecord.test.js

```diff
@@ -1,60 +1,50 @@
 import privateKeyMock from './fixtures/wallet.json' assert { type: 'json' };
 
 describe('setRecord', async () => {
-    const setRecordPath = './setRecord.js';
+    let arweaveMock;
+    let signMock;
+    let postMock;
+    let createTransactionMock;
 
-    let writeInteraction = mock.fn()
-    writeInteraction.mock.mockImplementationOnce(() => ({
-        type: 'ok',
-        state: {
-            ticker: 'antName',
-            name: 'arnsName',
-            owner: 'ownerAddress',
-            controller: 'ownerAddress',
-            evolve:null,
-            records: {
-                "@": {
-                    transactionId: 'validManifestId',
-                    ttlSeconds: 900
-                }
-            },
-            balances: {
-                owner_address: 1
-            }
-        }
+    createTransactionMock = mock.fn()
+    createTransactionMock.mock.mockImplementationOnce(() => ({
+        id: 'write-tx-id',
+        addTag: () => ({})
+    }))
+
+    postMock = mock.fn(async () => Promise.resolve({ 
+        status: 200, 
+        statusText: 'OK', 
+        data: {} 
     }));
-    
-    const warpContractsMock = {
-        WarpFactory: {
-            forMainnet: () => ({
-                pst: () => ({
-                    connect: () => ({
-                        writeInteraction: writeInteraction,
-                        dryWrite: mock.fn(async () => 'dryWriteMockResponse'),
-                    })
-                })
-            })
-        }
-    };
 
-    // Mock the Warp Contracts SDK
-    const { setRecord } = await esmock(setRecordPath, {
-        ['warp-contracts']: warpContractsMock
-    });
+    signMock = mock.fn()
 
+    arweaveMock = mock.method(arweave, 'init', () => {
+      return {
+        createTransaction: createTransactionMock,
+        transactions: {
+            post: postMock,
+            sign: signMock
+        },
+      }
+    })
+    
     it('should successfully return a dry write response', async () => {
-        const response = await setRecord('validAntAddress', 'validManifestId', privateKeyMock, true);
-        assert.ok(response);
-        assert.strictEqual(response, 'dryWriteMockResponse');
+        const { response, transactionId } = await setRecord('validAntAddress', 'validManifestId', privateKeyMock, true);
+        assert.strictEqual(response.status, 200);
+        assert.strictEqual(response.statusText, 'OK');
+        assert.strictEqual(transactionId, 'dry-run-tx-id')
     });
 
     it('should successfully returns a write interaction response', async () => {
-        const response = await setRecord('validAntAddress', 'validManifestId', privateKeyMock, false);
-        assert.ok(response);
-        assert.strictEqual(response.state.owner, 'ownerAddress');
-        assert.strictEqual(response.state.records['@'].transactionId, 'validManifestId');
+        const { response, transactionId } = await setRecord('validAntAddress', 'validManifestId', privateKeyMock, false);
```

Comment:
here you should check that the mocked methods have been called correctly by setRecord 

e.g. from upload.test.js
```
    assert.equal(getStatusMock.mock.callCount(), 2)
    assert.equal(getUploaderMock.mock.callCount(), 1)
    assert.deepEqual(getStatusMock.mock.calls[0].arguments, [txId])
    assert.deepEqual(getStatusMock.mock.calls[1].arguments, [txId])
```